packages/dds/merge-tree/src/test/mergeTreeOperationRunner.ts

```diff
@@ -29,6 +29,59 @@ export type TestOperation = (
 	random: IRandom,
 ) => IMergeTreeOp | undefined;
 
+export const insertField: TestOperation = (
+	client: TestClient,
+	opStart: number,
+	opEnd: number,
+	random: IRandom,
+) => {
+	const numberText = random.string(5, "0123456789");
+	if (
+		// start is not a number
+		Number.isNaN(Number(client.getText(opStart, opStart + 1))) &&
+		// previous is not a number
+		(opStart === 0 ||
+			opStart >= client.getLength() ||
+			Number.isNaN(Number(client.getText(opStart - 1, opStart))))
+	) {
+		return client.insertTextLocal(opStart, numberText);
+	}
+};
+
+const numberRange = (
+	client: TestClient,
+	start: number,
+): { startPos: number | undefined; endPos: number | undefined } => {
+	let startPos = start;
+	while (startPos > 0 && !Number.isNaN(Number(client.getText(startPos, startPos + 1)))) {
```

Comment:
don't you need to check is both directions?

Reply:
oh. i see. you depend on the format, so you find the start and know the end. this still has the problem of not being able to find fields which are next to one another, you will only even find the first one.

Reply:
you should add some comments here, and where the fields are generated, as this code is tightly coupled, and if either side changes the test could break